wikidict/render.py

```diff
@@ -0,0 +1,386 @@
+        sections = [sections]  # type: ignore
+    for section in sections:
+        etyl_data = parsed_sections.pop(section, [])
+        if etyl_data:
+            etymology = find_etymology(word, locale, etyl_data[0])
+
+    definitions = find_definitions(word, parsed_sections, locale)
+
+    if definitions or force:
+        prons = find_pronunciations(code, pronunciation[locale])
+        nature = find_genre(code, genre[locale])
+
+    # find if variant and delete unwanted definitions
+    variants = []
+    if locale == "fr" and not definitions:  # Pure verb form, no definition
+        for title, s in parsed_sections.items():
+            if title.startswith("{{S|verbe|fr|flexion"):
+                for t in s[0].templates:
+                    if t.__str__().startswith("{{fr-verbe-flexion"):
+                        infinitive = process_templates(word, clean(t.__str__()), locale)
+                        variants.append(infinitive)
+
+    return Word(prons, nature, etymology, definitions, variants)
+
+
+def get_latest_json_file(locale: str, output_dir: Path) -> str:
+    """Get the name of the last data-*.json file."""
+    files = os.listdir(output_dir)
+    files = [f for f in files if f.startswith("data_wikicode-") and f.endswith(".json")]
+    if not files:
+        return ""
+    files = sorted(files)
+    return files[-1]
+
+
+def load(input_file: Path) -> Dict[str, str]:
+    """Load the big JSON file containing all words and their details."""
+    with input_file.open(encoding="utf-8") as fh:
+        words: Dict[str, str] = json.load(fh)
+    print(f">>> Loaded {len(words):,} words from {input_file}", flush=True)
+    return words
+
+
+def debug_render(in_words: Dict[str, str], locale: str) -> None:
+    sections = defaultdict(list)
+    templates: Dict[str, str] = {}
+    for in_word, code in in_words.items():
+        for title in find_titles(code, locale):
+            sections[title].append(in_word)
+
+        parsed_sections = find_sections(code, locale)
+        defs = "\n".join(str(s) for s in parsed_sections.values())
+        for template in re.findall(r"({{[^{}]*}})", defs):
+            template = template.split("|")[0].lstrip("{").rstrip("}").strip()
+            if template not in templates:
+                templates[template] = in_word
+
+    with open("sections.txt", "w") as f:
+        for title, entries in sorted(sections.items()):
+            f.write(f"{title!r} ({len(entries):,})\n")
+            if len(entries) < 10:
+                # Most likely errors/mispellings
+                for entry in entries:
+                    f.write(f"    - {entry!r}\n")
+            else:
+                f.write(f"    - {entries[0]!r}\n")
+
+    if templates:
+        with open("templates.txt", "w") as f:
+            for template, entry in sorted(templates.items()):
+                f.write(f"{entry!r} => {template!r}\n")
+
+
+def render(in_words: Dict[str, str], locale: str) -> Words:
```

Comment:
This is here we should try to parallelize stuff. For now, let's do keep it like that, the improvement can be done later.